### PR TITLE
fix: handle multi dimensional flavors on patch android command

### PR DESCRIPTION
**Pull request by erickzanardo**
## Description

Change the patch command to handle multi dimensional flavors on `patch android` command.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart [hunk 1]
```diff
@@ -118,108 +119,118 @@ If this option is not provided, the version number will be determined from the p
...
-      releaseVersion = argReleaseVersion;
-    } else {
-      logger.detail('No release version provided. Determining from bundle.');
-      final flutterVersionString =
-          await shorebirdFlutter.getVersionAndRevision();
-      final buildProgress = logger.progress(
-        'Building patch with Flutter $flutterVersionString',
-      );
+
+    return runScoped(() async {
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart:123]: I'm not clear why we're adding an extra runScoped here – it doesn't look like we're overriding any dependencies?

File: packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart [hunk 2]
```diff
@@ -118,276 +119,295 @@ If this option is not provided, the version number will be determined from the p
...
-      },
-    );
+      } on ArtifactNotFoundException catch (error) {
+        logger.err(error.toString());
+        return ExitCode.software.code;
+      } on MultipleArtifactsFoundException catch (error) {
+        logger.err(error.toString());
+        return ExitCode.software.code;
+      }
+    });
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart:411]: I think you may still be working on this, but if this logic stays in the patch command, it should probably be more local to the build commands, even if it results in some duplicated code.

Comment (erickzanardo) [reply packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart:411]: yeah, I uploaded this PR, which I guess made patch command working. But I am working in parallel in a better version of it, that implements this as we discussed internally

**Comment by erickzanardo** [comment]
closing in favor of #1968

**Pull request closed without merging**